filter_plugins/merge.py

```diff
@@ -51,8 +52,26 @@ def merge_with_app(base, new):
             if new_service.get(k) is None:
                 continue
             if isinstance(new_service[k], list):
-                # Use the list(set()) trick to remove duplicated items
-                base_service[k] = sorted(list(set(new_service[k] + base_service[k])))
+                new_service_files = [Path(f).name for f in new_service[k]]
+                # Use the list(set()) trick to remove duplicated items. We also
+                # filter files based on their name, allowing to override
+                # templates for an app/service.
+                #
+                # Example: while merging baz/foo.yml (new) and bar/foo.yml (old)
+                # as they both share the same file name, e.g. foo.yml, the new
+                # one will be used and the old one ignored.
+                base_service[k] = sorted(
+                    list(
+                        set(
+                            new_service[k]
+                            + [
+                                f
+                                for f in base_service[k]
+                                if Path(f).name not in new_service_files
```

Comment:
not sure how they could be duplicates since you are checking that the file name if no in new_service[k].

Reply:
True. There cannot be duplicates anymore.